proxy/src/main/java/com/velocitypowered/proxy/connection/client/ClientConfigSessionHandler.java

```diff
@@ -120,24 +120,31 @@ public boolean handle(final PluginMessagePacket packet) {
       // but at this time the backend server may not be ready
     } else if (serverConn != null) {
       serverConn.ensureConnected().write(packet.retain());
+    } else {
+      player.getConnection().write(packet.retain());
     }
     return true;
   }
 
   @Override
   public boolean handle(PingIdentifyPacket packet) {
-    if (player.getConnectionInFlight() != null) {
-      player.getConnectionInFlight().ensureConnected().write(packet);
+    final VelocityServerConnection serverConn = player.getConnectionInFlight();
+    if (serverConn != null) {
+      serverConn.ensureConnected().write(packet);
+    } else {
+      player.getConnection().write(packet);
```

Comment:
isn't this sending data to the wrong thing? you either want to get the connection in flight, or the connected server?